File: src/main/java/hr/tvz/thesis/bookstore/domain/Order.java
package hr.tvz.thesis.bookstore.domain;

import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.List;
import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.JoinTable;
import javax.persistence.ManyToMany;
import javax.persistence.ManyToOne;
import javax.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
@Table(name = "Orders")
public class Order {

  @Id
  @GeneratedValue(strategy = GenerationType.IDENTITY)
  private Long id;

  @Column(name = "date_placed")
  private LocalDateTime datePlaced;

  @Column(name = "total_price")
  private BigDecimal totalPrice;

  @ManyToOne
  @JoinColumn(name = "user_id")
  private User user;

  @ManyToMany
  @JoinTable(
      name = "orders_book",
      joinColumns = { @JoinColumn(name = "order_id") },
      inverseJoinColumns = { @JoinColumn(name = "book_id") }
  )
  private List<Book> books;
}

File: src/main/bookstore-angular/src/app/domain/order.ts
import {Book} from "./book";
import {User} from "./user";

export class Order {

  constructor(id: number, datePlaced: Date, totalPrice: number, user: User, books: Book[]) {
    this.id = id;
    this.datePlaced = datePlaced;
    this.totalPrice = totalPrice;
    this.user = user;
    this.books = books;
  }

  id: number;
  datePlaced: Date;
  totalPrice: number;
  user: User;
  books: Book[];
}
